web-console/src/views/task-view/tasks-view.tsx

```diff
@@ -208,7 +210,16 @@ ORDER BY "rank" DESC, "created_time" DESC`;
       showTerminateAllSupervisors: false,
 
       tasksLoading: true,
-      taskFilter: props.taskId ? [{ id: 'task_id', value: props.taskId }] : [],
+      taskFilter: props.taskId
+        ? props.datasourceId
+          ? [
+              { id: 'task_id', value: props.taskId },
```

Comment:
@vogievetsky I realize this is probably a bed way to do this, where should this actually live

Reply:
probably in the constructor